StripeFinancialConnections/StripeFinancialConnections/Source/Native/PartnerAuth/PartnerAuthViewController.swift

```diff
@@ -637,6 +634,28 @@ final class PartnerAuthViewController: SheetViewController {
                 pane: .partnerAuth
             )
     }
+
+    // Defensive programming to avoid completing the same auth session twice.
+    //
+    // There's been a series of odd edge-cases where the same auth session
+    // could complete twice, so this acts as future-proofing that this
+    // never happens again.
+    private var lastCompletedAuthSessionId: String?
+    private func didComplete(withAuthSession authSession: FinancialConnectionsAuthSession) {
+        if lastCompletedAuthSessionId != authSession.id {
+            lastCompletedAuthSessionId = authSession.id
+            delegate?.partnerAuthViewController(self, didCompleteWithAuthSession: authSession)
```

Comment:
this is very important to double check: this code should ensure that `didCompleteWithAuthSession` DOES get called but only once per an `authSession` 

_(a user can have multiple auth sessions per one "link account session" [or a session through the financial connections auth flow]...all they have to do is cancel their current browser session or go back to institution/bank picker screen)_